internal/pkg/manifest/env.go

```diff
@@ -41,6 +69,49 @@ type environmentVPCConfig struct {
 	Subnets subnetsConfiguration `yaml:"subnets,omitempty"`
 }
 
+func (cfg *environmentVPCConfig) loadVPCConfig(env *config.CustomizeEnv) {
+	if env.IsEmpty() {
+		return
+	}
+	if adjusted := env.VPCConfig; adjusted != nil {
+		cfg.loadAdjustedVPCConfig(adjusted)
+	}
+	if imported := env.ImportVPC; imported != nil {
+		cfg.loadImportedVPCConfig(imported)
+	}
+}
+
+func (cfg *environmentVPCConfig) loadAdjustedVPCConfig(vpc *config.AdjustVPC) {
+	cfg.CIDR = ipNetP(vpc.CIDR)
+	hasCustomAZNames := len(vpc.AZs) > 0
+	cfg.Subnets.Public = make([]subnetConfiguration, len(vpc.PublicSubnetCIDRs))
+	cfg.Subnets.Private = make([]subnetConfiguration, len(vpc.PrivateSubnetCIDRs))
+	for i, cidr := range vpc.PublicSubnetCIDRs {
+		cfg.Subnets.Public[i].CIDR = ipNetP(cidr)
+		if hasCustomAZNames {
```

Comment:
something like...
```suggestion
		if hasCustomAZNames && len(vpc.AZs) > i {
```
? just so it doesn't panic?

Reply:
done, replaced with `if len(vpc.AZs) > i` - we shouldn't need `hasCustomAZNames` then

tbh I think we're pretty safe 😛 it's good that we don't panic, and if there is a mistake in the generated manifest then `Validate` will catch it before deploying:
https://github.com/aws/copilot-cli/blob/e92a4bcd20856e4836a59337cc5b6bed1d214e46/internal/pkg/manifest/validate_env.go#L86-L127